avm/res/sql/server/main.bicep

```
@@ -799,3 +798,96 @@ type elasticPoolPropertyType = {
   @description('Optional. Whether or not this elastic pool is zone redundant, which means the replicas of this elastic pool will be spread across multiple availability zones.')
   zoneRedundant: bool?
 }
+
+@export()
+type encryptionProtectorType = {
+  @description('Required. The name of the server key.')
+  serverKeyName: string
+
+  @description('Optional. The encryption protector type like \'ServiceManaged\', \'AzureKeyVault\'.')
+  serverKeyType: 'ServiceManaged' | 'AzureKeyVault'?
+
+  @description('Optional. Key auto rotation opt-in flag. Either true or false.')
+  autoRotationEnabled: bool?
+}
+
+@export()
+type vulnerabilityAssessmentType = {
+  @description('Required. The name of the vulnerability assessment.')
+  name: string
+
+  @description('Optional. The recurring scans settings.')
+  recurringScans: recurringScansType?
+
+  @description('Required. The resource ID of the storage account to store the scan reports.')
+  storageAccountResourceId: string
+
+  @description('Optional. Specifies whether to use the storage account access key to access the storage account.')
+  useStorageAccountAccessKey: bool?
+
+  @description('Optional. Specifies whether to create a role assignment for the storage account.')
+  createStorageRoleAssignment: bool?
+}
+
+@export()
+type firewallRuleType = {
+  @description('Required. The name of the firewall rule.')
+  name: string
+
+  @description('Optional. The start IP address of the firewall rule. Must be IPv4 format. Use value \'0.0.0.0\' for all Azure-internal IP addresses.')
+  startIpAddress: string?
+
+  @description('Optional. The end IP address of the firewall rule. Must be IPv4 format. Must be greater than or equal to startIpAddress. Use value \'0.0.0.0\' for all Azure-internal IP addresses.')
+  endIpAddress: string?
+}
+
+@export()
+type keyType = {
+  @description('Optional. The name of the key. Must follow the [<keyVaultName>_<keyName>_<keyVersion>] pattern.')
+  name: string?
+
+  @description('Optional. The server key type like \'ServiceManaged\', \'AzureKeyVault\'.')
```

```suggestion
  @description('Optional. The server key type.')
```
See previous comment.